packages/editor-ui/src/stores/settings.ts

```diff
@@ -238,11 +238,8 @@ export const useSettingsStore = defineStore(STORES.SETTINGS, {
 		setPromptsData(promptsData: IN8nPrompts): void {
 			Vue.set(this, 'promptsData', promptsData);
 		},
-		setAllowedModules(allowedModules: { builtIn?: string; external?: string }): void {
-			this.settings.allowedModules = {
-				...(allowedModules.builtIn && { builtIn: allowedModules.builtIn.split(',') }),
-				...(allowedModules.external && { external: allowedModules.external.split(',') }),
-			};
+		setAllowedModules(allowedModules: { builtIn?: string[]; external?: string[] }): void {
```

Comment:
what did this break? can we add a test here to catch whatever broke?

Reply:
[we updated the server-side code](https://github.com/n8n-io/n8n/pull/6055/files#diff-e1c41b1f8cc627242377b30fbee7af28472c81abe5f4f41da1c513f264f128e2R316-R317), that broke this because `allowedModules.builtIn` is a `string[]` now and not `string`.

This could have been caught if we set `NODE_FUNCTION_ALLOW_EXTERNAL` for smoke tests.

Reply:
why not allow that?

Reply:
because I'm trying to finish other tasks right now  , and just wanted to get the frontend working for anyone who has that env variable set.
If you prefer, I can set this draft, and add some checks on monday.

Reply:
👍🏽  whatever works.. we should protect against this again.. maybe @flipswitchingmonkey can help here as well

Reply:
this broke because frontend and backend were using different interfaces for the same class and thus 'hid' the different implementation (unsplit-string vs split string). sadly this did not catch the parameter in setAllowedModules :(

Reply:
yes ideally it would have been caught by our tests

Reply:
pushed a commit with just the config variable set for e2e tests, and [the CI failed](https://github.com/n8n-io/n8n/actions/runs/4799426715/jobs/8539143833).
added the fix, and now [the CI passes](https://github.com/n8n-io/n8n/actions/runs/4799501516).